### ISSUE TITLE
Ephemeral rooms

### ISSUE THREAD

**Issue by bcpierce00** [clowdr-app/clowdr#69]
Ephemeral rooms got lost somewhere along the way.  I think this should be the default (to prevent too many rooms from getting created and not garbage collected), or -- perhaps even better -- when you're creating a room you should be offered a choice of
- permanent (perhaps this requires special admin privileges?)
- delete at a given time
- delete when no one has been in the room for 10 minutes

**Comment by EdNutting** [comment]
@rossng I'm wondering what the best way to do this is. Adding more fields to the room state (to handle ephemeral status) seems undesirable. In addition, we already have an issue with the Cron job setup for managing transitions within a room/livestream. 

Should we add a queue table that holds the list of "transitions to be applied" in a room/livestream, and execute that queue in the actions service every X seconds/minutes? This would potentially avoid us generating lots of cron tasks when doing prepare-broadcasts, and also enable us to automatically maintain the list (which prepare-broadcasts is currently handling manually and that we can't delete from because of how the cron tasks work). Such a queue would also make implementing Ephemeral rooms and arbitrary time-limited rooms (as opposed to the existing shuffle rooms system) much easier?

**Comment by rossng** [comment]
I don't think we should unify the list of transitions (something I'm already trying to eliminate) between rooms and livestreaming. They don't overlap at all afaict.

I guess ephemeral rooms are always public? So we could just add them as a `roomPrivacyName`.
